### PR TITLE
[UX] fixed full sync stop on delete

### PR DESCRIPTION
**Pull request by julianseeger**
When stopping a sync currently running, the UX was really bad. The ongoing transfers and even new ones of the pending sync needed to get executed before the sync was really removed. Now, the sync is removed "instantly" from the UI, the sync is stopped asynchronously and no new events will get in.

(The tests should be clear about what went wrong in background and the "instantly" ui stuff was resolved by starting a non-ui thread)

### CODE REVIEW

File: src/main/java/de/qabel/desktop/ui/sync/item/SyncItemController.java [hunk 1]
```diff
@@ -144,17 +144,19 @@ public void delete() {
...
+                            boxSyncRepository.delete(syncConfig);
+                            new Thread(() -> {
+                                if (syncConfig.getSyncer() != null) {
+                                    try {
+                                        syncConfig.getSyncer().stop();
+                                    } catch (InterruptedException e) {
+                                        // best effort
+                                        alert("error while stopping sync: " + e.getMessage(), e);
+                                    } finally {
+                                        syncConfig.setSyncer(null);
```
Comment (audax) [inline src/main/java/de/qabel/desktop/ui/sync/item/SyncItemController.java]: You spawn a new thread and then stop a syncer there? Why is there a syncer in a new thread?

File: src/main/java/de/qabel/desktop/daemon/management/DefaultTransferManager.java [hunk 2]
```diff
@@ -44,7 +44,7 @@ public void addDownload(Download download) {
 
     @Override
     public List<Transaction> getHistory() {
-        return Collections.unmodifiableList(history);
+        return history;
```
Comment (audax) [inline src/main/java/de/qabel/desktop/daemon/management/DefaultTransferManager.java]: Uhm...why the change?

File: src/main/java/de/qabel/desktop/ui/sync/item/SyncItemController.java [hunk 3]
```diff
@@ -144,17 +144,19 @@ public void delete() {
...
-                                    syncConfig.setSyncer(null);
-                                    boxSyncRepository.delete(syncConfig);
+                            boxSyncRepository.delete(syncConfig);
+                            new Thread(() -> {
+                                if (syncConfig.getSyncer() != null) {
+                                    try {
+                                        syncConfig.getSyncer().stop();
+                                    } catch (InterruptedException e) {
+                                        // best effort
+                                        alert("error while stopping sync: " + e.getMessage(), e);
```
Comment (audax) [inline src/main/java/de/qabel/desktop/ui/sync/item/SyncItemController.java]: Does alert() work from a different thread?

File: src/test/java/de/qabel/desktop/daemon/sync/worker/DefaultSyncerTest.java [hunk 4]
```diff
@@ -128,16 +140,34 @@ public void stopStopsTransactions() throws Exception {
...
+        syncer = new DefaultSyncer(config, volume, manager);
+        syncer.run();
+
+        waitUntil(() -> manager.getTransactions().size() == 1 && syncer.isPolling());
+        syncer.stop();
+        manager.getHistory().clear();
+
+        nav.pushNotification(new BoxFile("a", "b", "c", 1L, 2L, new byte[0]), CREATE);
+        // wait for something to not happen?
+        Thread.sleep(100);
```
Comment (audax) [inline src/test/java/de/qabel/desktop/daemon/sync/worker/DefaultSyncerTest.java]: Can't you just waitUntil() manage.getHistory is empty?

File: src/main/java/de/qabel/desktop/daemon/management/DefaultTransferManager.java (hunk 2 above)
Comment (julianseeger) [inline src/main/java/de/qabel/desktop/daemon/management/DefaultTransferManager.java]: to make the test more readable and because I don't know why I wrapped it unmodifiable.
The test can now call `manager.getHistory().clear();` and assert later that it is empty instead of asserting one element in the history (that is always there when you sync something because it's the root folder of the sync)

Comment (audax) [inline src/main/java/de/qabel/desktop/daemon/management/DefaultTransferManager.java]: Makes sense.

File: src/main/java/de/qabel/desktop/ui/sync/item/SyncItemController.java (hunk 3 above)
Comment (julianseeger) [inline src/main/java/de/qabel/desktop/ui/sync/item/SyncItemController.java]: yep, alert checks weather it's on the Platform thread or not ... and is damn broken, wait...

File: src/test/java/de/qabel/desktop/daemon/sync/worker/DefaultSyncerTest.java (hunk 4 above)
Comment (julianseeger) [inline src/test/java/de/qabel/desktop/daemon/sync/worker/DefaultSyncerTest.java]: At that point it is always empty. the `pushNotification` method calls a `notifyAsync` (to allow for navigation refreshes to happen even if the observers are slow) and that is asynchronously scheduled to the TransferManager ... and then added to the history. so a waitUntil(manage.getHisotry()::isEmpty) would be true without the fix
